svc.go

```diff
@@ -46,7 +48,7 @@ func newAutoNATService(ctx context.Context, c *config) (*autoNATService, error)
 		reqs:   make(map[peer.ID]int),
 	}
 
-	if c.forceServer {
+	if c.forceReachability {
```

Comment:
Can we add a comment stating _why_ this is fine? Specifically, we won't construct a service if we're (a) forcing reachability and (b) aren't actually reachable.

Reply:
moved out to autonat so that this constructor does less potentially surprising things.